sdk/keyvault/Azure.Security.KeyVault.Certificates/src/LightweightPkcs8Decoder.cs

```diff
@@ -0,0 +1,192 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.IO;
+using System.Reflection;
+using System.Runtime.CompilerServices;
+using System.Security.Cryptography;
+
+namespace Azure.Core
+{
+    internal partial class LightweightPkcs8Decoder
+    {
+        private static readonly byte[] s_derIntegerOne = { 0x02, 0x01, 0x01 };
+
+        private static readonly byte[] s_ecAlgorithmId =
+        {
+            0x06, 0x07, 0x2A, 0x86, 0x48, 0xCE, 0x3D, 0x02, 0x01, // OID 1.2.840.10045.2.1, id-ecPublicKey
+        };
+
+        public static ECDsa DecodeECDsaPkcs8(byte[] pkcs8Bytes)
```

Comment:
My official recommendation is to delete all of this code.

On .NET Framework (Windows-only) (and on .NET Core 2.1 on Windows, if you wish) you can use

```C#
using (CngKey cngKey = CngKey.Import(pkcs8Bytes, CngKeyBlobFormat.Pkcs8PrivateKeyBlob))
{
    return new ECDsaCng(cngKey);
}
```

.NET Core 3.1+ for all OSes can use `ECDsa,ImportPkcs8PrivateKey`.

For .NET Core 2.1 (at least on Linux and macOS), I recommend that you not support this scenario.  .NET Core 2.1 goes out of support in August (2021).  The manual ASN.1 code only applies to at most 6% of the .NET (nee Core) userbase, per https://dotnet.microsoft.com/platform/telemetry.  To me, that's too much risk for too little reward.

Reply:
We are discussing that, but a final decision may not come for some time. I do want to pull most of this code out in favor of .NET implementations. Hopefully most of this is temporary.

Reply:
Resolving this comment since it's already tracked in #19196